src/ctapipe/image/extractor.py

```diff
@@ -424,6 +444,12 @@ def __call__(
         charge, peak_time = extract_around_peak(
             waveforms, 0, waveforms.shape[-1], 0, self.sampling_rate_ghz[tel_id]
         )
+
+        # reduce dimensions for gain selected data to (n_pixels, )
+        if selected_gain_channel is not None:
```

Comment:
In case of gain selected data, you only have a single image, but due to the changes made here to support also non gain selected data, it will have shape `(1, n_pixels)`, we throw away that outer dimension to stay backwards compatible (1d array for dl1 images in case of gain selection).